cuqi/sampler/_conjugate_approx.py

```diff
@@ -0,0 +1,52 @@
+from cuqi.distribution import Posterior, Laplace_diff, Gamma
+import numpy as np
+import scipy as sp
+
+class ConjugateApprox:
+    """ Approximate Conjugate sampler
+
+    Sampler for sampling a posterior distribution where the likelihood and prior can be approximated
+    by a conjugate pair.
+
+    Currently supported pairs are:
+    - (Laplace_diff, Gamma): Approximated by (GaussianPrec, Gamma)
```

Comment:
I am unsure about the idea of a single (approximate) conjugate sampler class. For two reasons I think. First, will this not amount to essentially a long if-else covering all the supported, and more or less unrelated, pairs? As such would seem splitting in dedicated classes might make sense?  Second, and I am unsure here, it seems that from a choice of a pair of DISTRIBUTIONS, we are choosing a conjugate SAMPLER, whereas it would seem more intuitive to be selecting conjugate DISTRIBUTIONS (though of course I do realize that the place in which this currently happens is where a name of a sampler is expected).

Reply:
#35